pip/tests/test_noisy_simulator.py

```diff
@@ -13,6 +13,17 @@
 
 # Tests for the Q# noisy simulator.
 
+
+def test_matrices_are_not_transposed_when_returned_back_to_python():
+    """
+    This test is due to the performance optimization we make in
+    `noisy_simulator/src/operation.rs/Operation::new`, we want to check that
+    we are reversing the transpose we made there before returning to Python.
+    """
+    op = Operation([[[1, 2], [3, 4]]])
```

Comment:
If you ever implement validity checks in Operation constructor, this may stop working. Operators should preserve normalization.

Reply:
Thank you, I used one of the Kraus operators of the depolarizing channel instead.